internal/provider/utils/utils.go

```diff
@@ -12,3 +16,14 @@ func NamespacedName(obj client.Object) types.NamespacedName {
 		Name:      obj.GetName(),
 	}
 }
+
+// Returns a partially hashed name for the string if it is more than 60 chars. Otherwise returns the original string
+func GetHashedName(name string) string {
+	if len(name) > 60 {
+		h := sha256.New() // Using sha256 instead of sha1 due to Blocklisted import crypto/sha1: weak cryptographic primitive (gosec)
+		hsha := h.Sum([]byte(name))
+		hashedName := strings.ToLower(fmt.Sprintf("%x", hsha))
+		return fmt.Sprintf("%s-%s", name[0:32], hashedName[0:16])
```

Comment:
7/8 nibbles should be for the short hash imho https://git-scm.com/book/en/v2/Git-Tools-Revision-Selection#Short-SHA-1